tests/plugins/colordialog/_helpers/tools.js

```diff
@@ -15,15 +15,44 @@ function assertColor( editor, inputColor, outputColor ) {
 	wait();
 }
 
+function assertSettingAndGettingColor( editor, options ) {
+	var toolbarButton = editor.ui.get( options.button );
+
+	bender.tools.setHtmlWithSelection( editor, '<p>Whenever [something] happens.</p>' );
+
+	editor.once( 'dialogShow', function( evt ) {
+		resume( function() {
+			var dialog = evt.data;
+
+			dialog.setValueOf( 'picker', 'selectedColor', options.inputColor );
+			dialog.getButton( 'ok' ).click();
+
+			assertValueFromDialog( editor, options );
+		} );
+	} );
+
+	toolbarButton.click( editor );
+	openColorDialog( toolbarButton );
```

Comment:
We could replace some of one-liner functions here with helpers to keep everything on the same level of abstraction - we'd have more readable code like this: 

```suggestion
	openColorPanel( toolbarButton, editor );
	openColorDialog( toolbarButton );
	
	function openColorPanel( button, editor ) {
		button.click( editor );
	}
```

We could do the same with e.g. `dialog.setValueOf( 'picker', 'selectedColor', options.inputColor );` -> `setColorInDialog( options.inputColor );` etc. But I'm not sure if creating a bunch of one-liners for such a purpose is justified, and the need of adding `editor` as a second argument also discourages me. WDYT?

Reply:
I really like this suggestion, but I have same doubts if we really need it…

But I'll do it anyway :smile: